### PR TITLE
CLDC-3318: Add maximum percentage discount validation for bulk uploads

### PR DESCRIPTION
**Pull request by RobertSullivanSoftwire**

### CODE REVIEW

File: config/locales/en.yml [hunk 1]
```diff
@@ -211,6 +211,7 @@ en:
         already_added: "You have already added this managing agent"
       merged: "That organisation has already been merged. Select a different organisation."
     not_answered: "You must answer %{question}"
+    range: "%{question} should be between %{min} and %{max}"
```
Comment (natdeanlewissoftwire) [inline config/locales/en.yml]: we've got validations.numeric.within_range that might be usable instead in this case (it has slightly different phrasing though), your call :)

File: app/services/bulk_upload/sales/year2024/row_parser.rb [hunk 2]
```diff
@@ -374,6 +374,15 @@ class BulkUpload::Sales::Year2024::RowParser
             },
             on: :after_log
 
+  validates :field_116,
+            numericality: {
```
Comment (natdeanlewissoftwire) [inline app/services/bulk_upload/sales/year2024/row_parser.rb:378]: nice! 🛤️

File: spec/services/bulk_upload/sales/year2024/row_parser_spec.rb [hunk 3]
```diff
@@ -725,6 +725,24 @@ def purch_id_field
       end
     end
 
+    describe "#field_116" do # percentage discount
```
Comment (natdeanlewissoftwire) [inline spec/services/bulk_upload/sales/year2024/row_parser_spec.rb:728]: can we add a test for <0% too?

**Review by natdeanlewissoftwire — APPROVED** [review]
lgtm!

File: config/locales/en.yml (hunk 1 above)
Comment (RobertSullivanSoftwire) [reply config/locales/en.yml]: Eh, I'm not going to quibble over should and must - let's get rid of this.

**Comment by kosiakkatrina** [comment]
I don't think this needed to be added separately, it should be getting caught by validate_numeric_min_max (?)

**Pull request merged**